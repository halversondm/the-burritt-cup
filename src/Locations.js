import React, { Component } from 'react';
import Container from "react-bootstrap/Container";
import Row from "react-bootstrap/Row";
import Col from 'react-bootstrap/Col';
import Post from './Post';

class Locations extends Component {

    constructor(props) {
        super(props);
        this.state = {
            locations: [
                {
                    "title": "",
                    "subtitle": "",
                    "metatitle": "",
                    "images": [
                        {
                            "location": "",
                            "title": "",
                            "alt": ""
                        }
                    ]
                }
            ]
        }
    }

    componentWillMount() {
        fetch('data/locations.json')
            .then(response => response.json())
            .then(json => this.setState({ locations: json }));
    }

    render() {
        return (
            <Container>
                <Row className="justify-content-center">
                    <Col lg={8} md={10}>
                        <h1>Locations</h1>
                    </Col>
                </Row>
                {this.state.locations.map((location, i) => {
                    return (
                        <Post key={i} title={location.title} subtitle={location.subtitle} metatitle={location.metatitle} images={location.images} summary={location.summary} awards={location.awards} />
                    );
                })}
            </Container>
        );
    }
}

export default Locations;