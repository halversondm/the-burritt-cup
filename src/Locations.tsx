import {useState, useEffect} from 'react';
import Container from "react-bootstrap/Container";
import Row from "react-bootstrap/Row";
import Col from 'react-bootstrap/Col';
import Post from './Post';

interface LocationsState {
    locations: Location[]
}

interface Location {
    title: string,
    subtitle: string,
    metatitle: string,
    awards: [],
    summary: string
    images: Image[]
}

interface Image {
    location: string
    title: string
    alt: string
}

export default function Locations() {

    const [state, setState] = useState(initialState);

    function initialState(): LocationsState {
        return {
            locations: [
                {
                    title: "",
                    subtitle: "",
                    metatitle: "",
                    awards: [],
                    summary: "",
                    images: [
                        {
                            location: "",
                            title: "",
                            alt: ""
                        }
                    ]
                }
            ]
        }
    }

    useEffect(() => {
        fetch('data/locations.json')
            .then(response => response.json())
            .then(json => setState({locations: json}));
    })

    return (
        <Container>
            <Row className="justify-content-center">
                <Col lg={8} md={10}>
                    <h1>Locations</h1>
                </Col>
            </Row>
            {state.locations.map((location, i) => {
                return (
                    <Post key={i} title={location.title} subtitle={location.subtitle} metatitle={location.metatitle}
                          images={location.images} summary={location.summary} awards={location.awards}/>
                );
            })}
        </Container>
    );
}
