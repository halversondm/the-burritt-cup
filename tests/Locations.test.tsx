import Locations from '../src/Locations';
import {render} from "@testing-library/react";

test("renders without crashing", () => {
  const unit = <Locations />;
  const component = render(unit);
  const tree = component.container;
  expect(tree.toString()).toMatchSnapshot();
});
