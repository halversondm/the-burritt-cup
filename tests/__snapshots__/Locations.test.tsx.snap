// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders without crashing 1`] = `"<div><div class="container"><div class="justify-content-center row"><div class="col-lg-8 col-md-10"><h1>Locations</h1></div></div><div><hr><div class="justify-content-center row"><div class="col-lg-8 col-md-10"><div class="post-preview"><h2 class="post-title"></h2><h3 class="post-subtitle"></h3><p class="post-meta"></p><p class="post-meta"><u>Photo Gallery</u></p><div class="carousel-root"><div class="carousel carousel-slider" style="width: 100%;"><ul class="control-dots"><li class="dot selected" value="0" role="button" tabindex="0" aria-label="slide item 1"></li></ul><button type="button" aria-label="previous slide / item" class="control-arrow control-prev control-disabled"></button><div class="slider-wrapper axis-horizontal"><ul class="slider animated" style="transform: translate3d(0,0,0); transition-duration: 350ms;"><li class="slide selected previous"><div><img alt=""><p class="legend"></p></div></li></ul></div><button type="button" aria-label="next slide / item" class="control-arrow control-next control-disabled"></button><p class="carousel-status">1 of 1</p></div><div class="carousel"><div class="thumbs-wrapper axis-vertical"><button type="button" class="control-arrow control-prev control-disabled" aria-label="previous slide / item"></button><ul class="thumbs animated" style="transform: translate3d(0,0,0); transition-duration: 350ms;"><li class="thumb selected" aria-label="slide item 1" style="width: 80px;" role="button" tabindex="0"><img alt=""></li></ul><button type="button" class="control-arrow control-next" aria-label="next slide / item"></button></div></div></div></div></div></div></div></div></div>"`;
